include/wx/window.h

```diff
@@ -1341,6 +1341,11 @@ class WXDLLIMPEXP_CORE wxWindowBase : public wxEvtHandler
     // send wxUpdateUIEvents to this window, and children if recurse is true
     virtual void UpdateWindowUI(long flags = wxUPDATE_UI_NONE);
 
+    // do the window-specific processing before processing the update event
+    // (mainly for deciding whether wxUpdateUIEvent::Is3State() is set)
+    virtual void DoPrepareUpdateWindowUI(wxUpdateUIEvent& event) const
+        { event.Allow3rdState(false); }
```

Comment:
Why bother doing this when it's the default anyhow?

Reply:
I have no good reason for this.  I just liked being explicit about the new state, especially since `IsCheckable()` defaults true, and this defaults false.  I can remove it if you prefer.

(As a separate statement, I think it's strange that `IsCheckable()` defaults true, but changing it would be a breaking change that doesn't seem justified.)